lib/Dialect/TritonGPU/Transforms/Pipeline.cpp

```diff
@@ -450,7 +450,7 @@ scf::ForOp LoopPipeliner::createNewForOp() {
   size_t depArgsBeginIdx = newLoopArgs.size();
   for (BlockArgument depArg : depArgs) {
     depArgsIdx[depArg] = newLoopArgs.size();
-    newLoopArgs.push_back(valueMapping[depArg][numStages - 2]);
+    newLoopArgs.push_back(valueMapping[depArg][numStages - 1]);
```

Comment:
I remember having to make this change here https://github.com/openai/triton/commit/660f2e8ccef376b1fa911ce4766568c8d1af4907#diff-6aa8da118d686382430e3b9ae7f7c1236e81e8a7598b28d76204fd5b5e43de7b. Does this PR still generates the same exact ptx as before for the dense matmul case?

Reply:
wait.. I think you are right. The generated ptx looks different. digging into it

Reply:
While you're at it, could you actually add a test that hashes the generated `ttgir` for some efficient V100/A100 config and make sure that it's equal to a reference hash that we will update manually? I think it would be super valuable!

Reply:
Sure.
I think I found a better instruction order, but I saw some extra barriers in the main loop. not sure what's happening